client/src/pages/Configurations/index.test.tsx

```diff
@@ -10,24 +18,116 @@ const renderPageView = () =>
     </BrowserRouter>
   );
 
-describe('Configurations', () => {
-  it('should contain a table with certain columns', async () => {
-    renderPageView();
-
-    expect(screen.getByTestId('table')).toBeInTheDocument();
-  });
-  it('should contain a call-to-action button', async () => {
+describe('Configurations Page', () => {
+  test('renders the Configurations page with title and search bar', () => {
     renderPageView();
     expect(
-      screen.getByText('Set up new condition', {
-        selector: 'button.usa-button[type=button]',
-      })
+      screen.getByText('Your reportable condition configurations')
     ).toBeInTheDocument();
-  });
-  it('should have a search component with correct placeholder', async () => {
-    renderPageView();
     expect(
       screen.getByPlaceholderText('Search configurations')
     ).toBeInTheDocument();
+    expect(
+      screen.getByRole('button', { name: 'Set up new condition' })
+    ).toBeInTheDocument();
+  });
+
+  test('opens the modal when "Set up new condition" button is clicked', async () => {
+    const user = userEvent.setup();
+    renderPageView();
+    const setUpButton = screen.getByRole('button', {
+      name: 'Set up new condition',
+    });
+    await user.click(setUpButton);
+    const modal = screen.getByRole('dialog');
+    expect(modal).toBeInTheDocument();
+    expect(getByText(modal, 'Set up new condition')).toBeInTheDocument();
```

Comment:
This can be reworked slightly to use Testing Library's `within` function:
```suggestion
    expect(within(modal).getByText('Set up new condition')).toBeInTheDocument();
```
This can be really useful when trying to narrow the search for the element, as you're doing here.

Reply:
ah thanks! I'll refactor a couple of calls that I have like this.